test/e2e/e2e_test.go

```diff
@@ -0,0 +1,119 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/loft-sh/vcluster/cmd/vclusterctl/cmd"
+	"github.com/loft-sh/vcluster/cmd/vclusterctl/flags"
+	"github.com/loft-sh/vcluster/cmd/vclusterctl/log"
+	logutil "github.com/loft-sh/vcluster/pkg/util/log"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestRunE2ETests(t *testing.T) {
+	gomega.RegisterFailHandler(ginkgo.Fail)
+	ginkgo.RunSpecs(t, "e2e suite")
+}
+
+var _ = ginkgo.Describe("e2e test", func() {
+	ginkgo.Context("E2E", func() {
+
+		ginkgo.BeforeEach(func() {
+		})
+
+		ginkgo.It("Creates VirtualCluster successfully", func() {
+			ctx := context.TODO()
+			ctrl.SetLogger(logutil.NewLog(0))
+			l := log.GetInstance()
+			scheme := runtime.NewScheme()
+
+			// run port forwarder and retrieve kubeconfig for the vcluster
+			vKubeconfigFile, err := os.CreateTemp(os.TempDir(), "vcluster_e2e_kubeconfig_")
+			if err != nil {
+				errMsg := fmt.Errorf("could not create a temporary file: %w", err)
+				ginkgo.Fail(errMsg.Error())
+			}
+			// TODO: cleanup
+			// vKubeconfigFile removal is done in the Framework.Cleanup() which gets called in ginkgo's AfterSuite()
+
+			connectCmd := cmd.ConnectCmd{
+				Log: l,
+				GlobalFlags: &flags.GlobalFlags{
+					Namespace: "vcluster",
+					Debug:     true,
+				},
+				KubeConfig: vKubeconfigFile.Name(),
+				LocalPort:  14550, // choosing a port that usually should be unused
+			}
+			err = connectCmd.Connect(ctx, "vcluster", nil)
+			if err != nil {
+				l.Fatalf("failed to connect to the vcluster: %v", err)
```

Comment:
In a previous step we're using ginkgo.Fail to abort/panic, here we're using l.Fatalf.
I think it'd make sense to just go with the ginkgo one, instead of mixing those here.

Reply:
makes sense, modified